include/odp/api/spec/crypto.h

```diff
@@ -710,11 +720,17 @@ typedef struct odp_crypto_op_param_t {
 
 	/** Offset from start of packet for hash result
 	 *
-	 *  Specifies the offset where the hash result is to be stored. In case
-	 *  of decode sessions, input hash values will be read from this offset,
-	 *  and overwritten with hash results. If this offset lies within
-	 *  specified 'auth_range', implementation will mute this field before
-	 *  calculating the hash result.
+	 *  In case of decode sessions, the expected hash will be read from
+	 *  this offset and compared with the calculated hash. After the
+	 *  operation the hash bytes will have undefined values.
+	 *
+	 *  In case of encode sessions the calculated hash will be stored in
+	 *  this offset.
+	 *
+	 *  If the hash_result_not_in_auth_range session parameter is false,
+	 *  the hash result location may overlap auth_range. In that case the
+	 *  result location will be zeroed in decode sessions before hash
+	 *  calculation. Zeroing is not done in encode sessions.
```

Comment:
Since hash_result_not_in_auth_range  applies only for decode sessions, I'd merge this last paragraph into decode and encode paragraphs above. So, that e.g. a person creating an encode session would need to care only on the paragraph starting with "In case of encode sessions..."

Reply:
Actually the paragraph, except the middle sentence, applies to encode sessions too. In encode sessions hash result may be in auth range but only if the ...not_in_auth_range parameter is not set. But in that case the zeroing is not done in encode sessions.